### ISSUE TITLE
Dataset problem: EC-Earth3-Veg, EC-Earth3

### ISSUE THREAD

**Issue by mwjury** [ESMValGroup/ESMValCore#835]
When doing an ensemble_statistics, metadata has to be the same between the single ensemble members. Depending on #673 for the error to appear. Currently tested for CMIP6 Amon TAS runs having data for this 5 experiments: historical, ssp126, ssp245, ssp370, ssp585.

This leads to problems with (luckily on 2 so far):
- [x]  EC-Earth3-Veg: #836
- [x]  EC-Earth3: #837

In the case of EC-Earth3-Veg and TAS there are differences in latitude points and bounds of -9.9475983e-14. So some simple rounding should do the trick.

```
iris.exceptions.MergeError: failed to merge into a single cube.
  Coordinates in cube.dim_coords differ: latitude.
```

- Project: CMIP6
- Full dataset description (fill out as many as you know, please):
  - dataset: EC-Earth3-Veg
  - ensemble: [r(1:4)i1p1f1]
  - experiment: [historical, ssp245]
  - mip: [CMIP, ScenarioMIP]
  - grid: gr
  - data version: [v20200225, v20200425]
  - variable used: tas
- Problems encountered (please describe what the actual problems are: e.g. wrong standard name, issue with dimensions):
- Pointer to existing copy of data on ESGF node (it would be very useful if you could provide a physical
fullpath to the file(s) that are causing the problem, e.g. on CEDA Jasmin or DKRZ):
CMIP/EC-Earth-Consortium/EC-Earth3-Veg/historical/r1i1p1f1/Amon/tas/gr/v20200225/tas_Amon_EC-Earth3-Veg_historical_r1i1p1f1_gr_201401-201412.nc
CMIP/EC-Earth-Consortium/EC-Earth3-Veg/historical/r4i1p1f1/Amon/tas/gr/v20200425/tas_Amon_EC-Earth3-Veg_historical_r4i1p1f1_gr_201401-201412.nc
or 
ScenarioMIP/EC-Earth-Consortium/EC-Earth3-Veg/ssp245/r1i1p1f1/Amon/tas/gr/v20200225/tas_Amon_EC-Earth3-Veg_ssp245_r1i1p1f1_gr_201501-201512.nc
ScenarioMIP/EC-Earth-Consortium/EC-Earth3-Veg/ssp245/r4i1p1f1/Amon/tas/gr/v20200425/tas_Amon_EC-Earth3-Veg_ssp245_r4i1p1f1_gr_201501-201512.nc
- Other notes and mentions:

**Please attach**
  - The recipe that you are trying to run, you can find a copy in the `run` directory in the output directory
[CMIP6_EC-Earth3-Veg_ensemble_statistic.yml.txt](https://github.com/ESMValGroup/ESMValCore/files/5400288/CMIP6_EC-Earth3-Veg_ensemble_statistic.yml.txt)

Cheers :beer:

**Comment by mwjury** [comment]
@valeriupredoi ensemble_statistics is opening new avenues for metadata discrepancies.

**Comment by mwjury** [comment]
Same thing happening with EC-Earth3.

Found between e.g. 
ScenarioMIP/EC-Earth-Consortium/EC-Earth3/ssp370/r4i1p1f1/Amon/tas/gr/v20200425/tas_Amon_EC-Earth3_ssp370_r4i1p1f1_gr_201601-201612.nc
ScenarioMIP/EC-Earth-Consortium/EC-Earth3/ssp370/r1i1p1f1/Amon/tas/gr/v20200310/tas_Amon_EC-Earth3_ssp370_r1i1p1f1_gr_201601-201612.nc

[CMIP6_EC-Earth3_ensemble_statistic.yml.txt](https://github.com/ESMValGroup/ESMValCore/files/5401524/CMIP6_EC-Earth3_ensemble_statistic.yml.txt)
